st2reactor/tests/unit/test_process_container.py

```diff
@@ -32,3 +35,34 @@ def test_no_sensors_dont_quit(self):
         self.assertEqual(process_container.stopped(), False)
         process_container.shutdown()
         process_container_thread.kill()
+
+    @patch.object(time, 'time', MagicMock(return_value=1439441533))
+    def test_dispatch_triggers_on_spawn_exit(self):
+        mock_dispatcher = Mock()
+        process_container = ProcessSensorContainer(None, poll_interval=0.1,
+                                                   dispatcher=mock_dispatcher)
+        sensor = {
+            'class_name': 'pack.StupidSensor'
+        }
+        process = Mock()
+        process_attrs = {'pid': 1234}
+        process.configure_mock(**process_attrs)
+        cmd = 'sensor_wrapper.py --class-name pack.StupidSensor'
+
+        process_container._dispatch_trigger_for_sensor_spawn(sensor, process, cmd)
+        mock_dispatcher.dispatch.assert_called_with(
+            'core.st2.sensor.process_spawn',
```

Comment:
That's test is fine for now and definitely a lot better than nothing, but in ideal world, we could some how test the whole pipeline.

(If we previously had a similar test, it probably wouldn't catch this issue since the person writing the test would probably incorrectly just check the name).